clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```
@@ -239,16 +251,49 @@ private void process(final ListOffsetsEvent event) {
     }
 
     /**
-     * Process event that indicates that the subscription changed. This will make the
+     * Process event that indicates that the subscription topics changed. This will make the
      * consumer join the group if it is not part of it yet, or send the updated subscription if
      * it is already a member.
      */
-    private void process(final SubscriptionChangeEvent ignored) {
+    private void process(final TopicSubscriptionChangeEvent event) {
+        if (subscriptions.subscribe(event.topics(), event.listener()))
```

this could throw errors I expect (ie.`IllegalStateException` if we mix subscription types), so should we just catch and completeExceptionally? (maybe generalize and do all actions except for the check manager.isPresent within a try, catch all Exceptions and completeExceptionally, to play safe)

---

Yes, add a try catch block for exceptions.